monitoring/tests/test_build_stats_collector.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+import unittest
+from unittest.mock import MagicMock
+
+from collectors.build_stats_collector import BuildStatsCollector, BuildStatsMissingError
+from tests.common import get_test_build_stats, EXAMPLE_TESTS_STATS
+
+
+class TestBuildStatsCollector(unittest.TestCase):
+    def setUp(self):
+        self.example_build_stats = get_test_build_stats()
+        self.example_test_stats = EXAMPLE_TESTS_STATS
+
+        self.example_build_stats_collector = MagicMock()
+        self.example_build_stats_collector.collect = MagicMock(return_value=self.example_build_stats)
+
+        self.example_test_stats_collector = MagicMock()
+        self.example_test_stats_collector.collect = MagicMock(return_value=self.example_test_stats)
+
+    def test_collecting_works(self):
+        collector = BuildStatsCollector(build_stats_collector=self.example_build_stats_collector,
+                                        test_stats_collector=self.example_test_stats_collector)
+        build_stats = collector.collect()
+        self.assertIsNotNone(build_stats)
+
+        self.assertEqual(build_stats.build_id, self.example_build_stats.build_id)
```

Comment:
Uncle Bob wouldn't be impressed... But we don't complain.
MK suggests helper assertion encapsulating these assertions.